core/src/main/java/com/novoda/espresso/TalkBackTestRule.java

```diff
@@ -27,16 +36,49 @@ public Statement apply(Statement base, Description description) {
         @Override
         public void evaluate() throws Throwable {
             talkBackStateSettingRequester.requestEnableTalkBack();
-            sleepToAllowTalkBackServiceToChangeState();
+            sleepUntil(talkBackIsEnabled());
 
             baseStatement.evaluate();
 
             talkBackStateSettingRequester.requestDisableTalkBack();
-            sleepToAllowTalkBackServiceToChangeState();
+            sleepUntil(talkBackIsDisabled());
+        }
+
+        private void sleepUntil(Condition condition) {
+            int retries = 0;
+            while (!condition.holds()) {
+                SystemClock.sleep(SLEEP_TO_WAIT_FOR_TALK_BACK_MILLIS);
+                if (retries == MAX_RETRIES_TO_WAIT_FOR_TALK_BACK) {
+                    throw talkBackToggleTimeOutError();
+                }
+                retries++;
+            }
+        }
+
+        private Condition talkBackIsEnabled() {
+            return new Condition() {
+                @Override
+                public boolean holds() {
+                    return !talkBackIsDisabled().holds();
```

Comment:
this is test land but creating a `Condition` instance every time the `holds` method is called seems a bit unnecessary?